controllers/nodehealthcheck_controller_test.go

```diff
@@ -931,28 +955,64 @@ var _ = Describe("Node Health Check CR", func() {
 				Expect(lease.Spec.RenewTime.Sub(lease.Spec.AcquireTime.Time) > 0).To(BeTrue())
 
 				// Wait for 2nd remediation to time out
-				time.Sleep(17 * time.Second)
+				Eventually(func(g Gomega) {
+					// get updated CR
+					g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(newCr), newCr)).To(Succeed())
+					g.Expect(cr.GetAnnotations()).To(HaveKeyWithValue(Equal("remediation.medik8s.io/nhc-timed-out"), Not(BeNil())))
+					g.Expect(newCr.GetName()).To(Equal(unhealthyNodeName))
+					g.Expect(newCr.GetAnnotations()).ToNot(HaveKey(Equal(annotationutils.NodeNameAnnotation)))
 
-				// get updated CR
-				Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(newCr), newCr)).To(Succeed())
-				Expect(cr.GetAnnotations()).To(HaveKeyWithValue(Equal("remediation.medik8s.io/nhc-timed-out"), Not(BeNil())))
+				}, time.Second*10, time.Millisecond*300).Should(Succeed())
 
 				// get updated NHC
-				Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
-				Expect(underTest.Status.UnhealthyNodes[0].Remediations[1].Resource.GroupVersionKind()).To(Equal(newCr.GroupVersionKind()))
-				Expect(underTest.Status.UnhealthyNodes[0].Remediations[1].TimedOut).ToNot(BeNil())
-				Expect(underTest.Status.Phase).To(Equal(v1alpha1.PhaseRemediating))
+				Eventually(func(g Gomega) {
+					g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+					g.Expect(underTest.Status.UnhealthyNodes[0].Remediations[1].Resource.GroupVersionKind()).To(Equal(newCr.GroupVersionKind()))
+					g.Expect(underTest.Status.UnhealthyNodes[0].Remediations[1].TimedOut).ToNot(BeNil())
+					g.Expect(underTest.Status.Phase).To(Equal(v1alpha1.PhaseRemediating))
+
+				}, time.Second*10, time.Millisecond*300).Should(Succeed())
 
-				//Incorrect lease expire time calculated with short remediation timeout instead of long one
-				leaseWrongExpireTimeShort := lease.Spec.AcquireTime.Time.Add(firstRemediationTimeout*3 + mockLeaseBuffer /*longest remediation timeout (5) multiply by tries (3) and added buffer*/)
+				// Wait for 3rd remediation to start
```

Comment:
There seems to be quite some code that can be put into a single function, at least for clarity

Reply:
Hmm, I'm really not sure.
IIUC Only trivial code are the blocks of: ` //Wait for 3rd remediation to start` and `// Wait for 4th remediation to start` which makes a little difference and I'm not sure helps with clarity.

Did you have something else in mind ?